Adds QA features to user docs

Shrinks99 opened the pull request:
Fixes #1695

### Changes
- Adds Crawl Review user docs
- Adds Quality Assurance section to the Archived Items page

### Caveats
We're getting to the point where it might be nice to start including images in some of these... 👀

Shrinks99 committed:
Update ReplayWeb.page Links

Shrinks99 committed:
QA initial guide commit

- Updates archived item page with the QA tab
- Adds review page and documents its features

Shrinks99 committed:
word improvements

Somebody told me once that I have the best words!

Shrinks99 committed:
Adds note about crawl analysis to the crawl settings tab

Shrinks99 committed:
Add note about general usefulness of analysis WACZs

ikreymer commented on docs/user-guide/archived-items.md:
```diff
@@ -22,19 +22,37 @@ Because <span class="status-warning">:bootstrap-x-octagon-fill: Canceled</span>
+
+Running crawl analysis will re-visit all pages within the archived item, comparing the data collected during analysis with the data collected during crawling. Crawl analysis runs with the same workflow limit settings used during crawling.
+
+Crawl analysis can be run multiple times, though it will only yield different results if the crawler, or [ReplayWeb.page](https://replayweb.page/docs/) — the software responsible for playback of archived web content — has been updated between runs. Analysis run data can be downloaded or deleted from the _Analysis Runs_ tab. While they are stored as WACZ files, analysis run WACZs only contain anal ...
```
```suggestion
Crawl analysis can be run multiple times, though results should only differ if the crawler version has been updated between runs. The analysis process is being constantly improved and future analysis runs should produce better results. Analysis run data can be downloaded or deleted from the _Analysis Runs_ tab. While they are stored as WACZ files, analysis run WACZs only contain analysis data and may not open correctly or be useful in other programs that replay archived content.
```

Shrinks99 replied:
Made some very minor changes to your suggestion for grammar.  Changed "QA workflow" to "analysis process" as we don't actually refer to this as "QA" in the app.

Do we want to completely exclude ReplayWeb.page updates as a variable here?  When replay bugs are fixed, that seems like the determining factor?  ...That said AFAIK it's built into the crawler now and so maybe this is a good change and it would be best to just list the version update for that to not complicate things?

ikreymer replied:
Yes, RWP is not changed individually but via crawler versions, so want to avoid potential confusion as to what needs to be updated specifically.

emma-sg commented on docs/user-guide/archived-items.md:
```diff
@@ -22,19 +22,37 @@ Because <span class="status-warning">:bootstrap-x-octagon-fill: Canceled</span>
 ## Archived Item Details
 
-The archived item details page is composed of five sections, though the Crawl Settings tab is only available for crawls and not uploads.
+The archived item details page is composed of six sections, though some tabs are only available for crawls and not uploads.
```
Will viewers see only some of these sections? Might be worth making a note of this if so

Shrinks99 replied:
QA page is hidden from viewers???  Not really sure why?  It currently has the pages list which is presumably useful for them to some extent, though all the actions on that page should be hidden from viewers.

ikreymer replied:
Yes, for now, since we don't have a way of making it read-only, since running assistive QA + manual review are not read-only actions. We may need a separate role that allows QA, in the future, but for now this should be alright.

Shrinks99 replied:
Well the pages list and rankings should ideally be viewable for viewers, but fair enough, not a high priority issue (IDK if we even have a single user with the viewer role on prod haha)

Perhaps something to fix in the next iterations?  I like the idea of a QA role!  Will update with a note.

SuaYoo replied:
What's the reason for needing to specify the number of sections? IMO specifying the number makes the count feel more important than it is, especially when it's dependent on whether QA is enabled, what the archived item type is, and maybe other factors in the future. How about something like:
```
Archived item details are organized into the following sections.
```
And then "(crawl-only)" next to the heading for each applicable subsection that describes the tab content.

emma-sg commented on docs/user-guide/archived-items.md:
```diff
@@ -22,19 +22,37 @@ Because <span class="status-warning">:bootstrap-x-octagon-fill: Canceled</span>
 
+### Quality Assurance
+
+The Quality Assurance tab displays a list of crawled pages, overall stats from collected through crawl analysis, and a list of all analysis runs. This tab is not available for uploaded archived items.
```
```suggestion
The Quality Assurance tab displays a list of crawled pages, overall stats collected through crawl analysis, and a list of all analysis runs. This tab is not available for uploaded archived items.
```

emma-sg commented on docs/user-guide/archived-items.md:
```diff
@@ -22,19 +22,37 @@ Because <span class="status-warning">:bootstrap-x-octagon-fill: Canceled</span>
+
+The Quality Assurance tab displays a list of crawled pages, overall stats from collected through crawl analysis, and a list of all analysis runs. This tab is not available for uploaded archived items.
+
+The pages list provides a record of all pages within the archived item and any ratings or notes assigned to the page during review. If analysis has been run, clicking on a page in the pages list will show that page in the review interface.
```
```suggestion
The pages list provides a record of all pages within the archived item, as well as any ratings or notes given to the page during review. If analysis has been run, clicking on a page in the pages list will go to that page in the review interface.
```

emma-sg commented on docs/user-guide/archived-items.md:
```diff
@@ -22,19 +22,37 @@ Because <span class="status-warning">:bootstrap-x-octagon-fill: Canceled</span>
+
+`Paid Feature`{ .badge-green }
+
+Like running a crawl workflow, running crawl analysis also uses execution time. Crawls and crawl analysis share the same concurrent crawling limit, however crawl analysis runs will be paused in favor of new crawls if the concurrent crawling limit is reached.
```
```suggestion
Like running a crawl workflow, running crawl analysis also uses execution time. Crawls and crawl analysis share the same concurrent crawling limit, but crawl analysis runs will be paused in favor of new crawls if the concurrent crawling limit is reached.
```

emma-sg commented on docs/user-guide/review.md:
```diff
@@ -0,0 +1,48 @@
+
+The Crawl Review page provides a streamlined interface for assessing the capture quality of pages within an archived item using the heuristics collected during crawl analysis.
+
+Crawls can only be reviewed once [crawl analysis](archived-items.md#crawl-analysis) has been run. If multiple analysis runs have been completed, the page analysis heuristics will be used from the selected analysis run, available beside the archived item name.
```
```suggestion
Crawls can only be reviewed once [crawl analysis](archived-items.md#crawl-analysis) has been run. If multiple analysis runs have been completed, the page analysis heuristics will be used from the selected analysis run, which are displayed next to the archived item name. The most recent analysis run is selected by default, but you can choose to display data from any other completed or stopped analysis run here as well.
```

emma-sg commented on docs/user-guide/review.md:
```diff
@@ -0,0 +1,48 @@
+
+### Extracted Text Comparison
+
+Text extracted during crawl analysis is compared to the extracted text created during crawling. Extracted text is compared on the basis of [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance).
```
```suggestion
Text extracted during crawl analysis is compared to the text extracted during crawling. Text is compared on the basis of [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance).
```

Shrinks99 replied:
Hmm, not sure I like this change, I think we need to fit "extracted" in at _least_ 3 more times before I'm truly happy with it.  ;)

emma-sg commented on docs/user-guide/review.md:
```diff
@@ -0,0 +1,48 @@
+!!! Tip "Caveats"
+    The number of resources may be higher on replay due to how components of ReplayWeb.page re-write certain request types. A discrepancy alone may not be an indicator that the page is broken, though generally it is a positive sign when the counts are equal.
+
+    Due to the complicated nature of resource count comparison, they are not available as sorting options in the pages list.
```
```suggestion
    Due to the complicated nature of resource count comparison, this is not available as a sorting option in the pages list.
```

emma-sg commented on docs/user-guide/review.md:
```diff
@@ -0,0 +1,48 @@
+
+## Page Review
+
+The pages list can be sorted using the analysis heuristics to determine the pages that are likely more important to review vs those that might require less attention. After selecting a page to review, looking over the analysis heuristics, and checking them against replay, make a decision about if the page capture was successful or unsuccessful and leave a note using the about what worked well or  ...
```
```suggestion
The pages list can be sorted using analysis heuristics to determine the pages that are likely more important to review versus those that might require less attention. After selecting a page to review, looking over the analysis heuristics, and checking them against replay, make a decision about if the page capture was successful or unsuccessful and leave a note about what worked well or what might be problematic.
```

Shrinks99 replied:
Yikes, embarassing.  Thanks for catching this one!

emma-sg commented on docs/user-guide/review.md:
```diff
@@ -0,0 +1,48 @@
+The pages list can be sorted using the analysis heuristics to determine the pages that are likely more important to review vs those that might require less attention. After selecting a page to review, looking over the analysis heuristics, and checking them against replay, make a decision about if the page capture was successful or unsuccessful and leave a note using the about what worked well or  ...
+
+??? Question "Should I review every page? (Spoiler alert: probably not!)"
+    When reviewing a crawl of a site that has many similar pages, all of which exhibit the same error and have similar heuristic scores, they're likely all like that and you can _probably_ save yourself the trouble. Depending on the website, the heuristic scores may not always be an accurate predictor of quality, but in our testing they are fairly consistent — consistency being the important fact ...
```
```suggestion
    When reviewing a crawl of a site that has many similar pages, all of which exhibit the same error and have similar heuristic scores, it's likely that they all are similarly broken, and you can _probably_ save yourself the trouble. Depending on the website, the heuristic scores may not always be an accurate predictor of quality, but in our testing they are fairly consistent — consistency being the important factor of this tool. It is up to you, the curator, to make the final quality judgement!
```

emma-sg commented on docs/user-guide/review.md:
```diff
@@ -0,0 +1,48 @@
+??? Question "Should I review every page? (Spoiler alert: probably not!)"
+    When reviewing a crawl of a site that has many similar pages, all of which exhibit the same error and have similar heuristic scores, they're likely all like that and you can _probably_ save yourself the trouble. Depending on the website, the heuristic scores may not always be an accurate predictor of quality, but in our testing they are fairly consistent — consistency being the important fact ...
+
+    Our recommended workflow is: run crawl analysis, examine the most severe issues as highlighted, examine some key examples of common layouts, review any other key pages, and score the crawl accordingly!
```
```suggestion
    Our recommended workflow is as follows: run crawl analysis, examine the most severe issues as highlighted, examine some key examples of common layouts, review any other key pages, and score the crawl accordingly!
```

emma-sg reviewed (changes requested):
Left a bunch of thoughts & suggestions!

Shrinks99 committed:
Update docs/user-guide/archived-items.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

Shrinks99 committed:
Update docs/user-guide/archived-items.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

Shrinks99 committed:
Update docs/user-guide/archived-items.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

Shrinks99 committed:
Update docs/user-guide/review.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

Shrinks99 committed:
Update docs/user-guide/review.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

Shrinks99 committed:
Update docs/user-guide/review.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

Shrinks99 committed:
Update docs/user-guide/review.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

Shrinks99 committed:
Update docs/user-guide/review.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

Shrinks99 committed:
Update docs/user-guide/review.md

Co-authored-by: Emma Segal-Grossman <hi@emma.cafe>

SuaYoo commented on docs/user-guide/archived-items.md:
```diff
@@ -22,19 +22,37 @@ Because <span class="status-warning">:bootstrap-x-octagon-fill: Canceled</span>
 
+### Quality Assurance
+
+The Quality Assurance tab displays a list of crawled pages, overall stats collected through crawl analysis, and a list of all analysis runs. This tab is not available for uploaded archived items.
```
This could lead more with how "list of crawled pages" and "overall stats" is relevant to a QA process. Maybe just by bringing up some of the keywords in the second paragraph earlier? Something like:

```diff
+ The Quality Assurance tab displays quality assurance data that is collected from manual reviews and analysis runs. This is where you can start new analysis runs, view quality metrics from older runs, and delete runs.

The pages list provides a record of all pages within the archived item, as well as any ratings or notes given to the page during review. If analysis has been run, clicking on a page in the pages list will go to that page in the review interface.
```

SuaYoo commented on docs/user-guide/review.md:
```diff
@@ -0,0 +1,48 @@
+
+### Screenshot Comparison
+
+Screenshots taken during crawl analysis are compared with the page thumbnail created during crawling. Screenshots are compared using [Pixelmatch](https://github.com/mapbox/pixelmatch).
```
Why note that we use Pixelmatch here?

Shrinks99 replied:
I think it's useful for folks to know how these scores are generated should they wish to test or develop similar thngs themselves.  Elsewhere in the docs we note the method behind text comparison, and that we use zxcvbn to derive password strength.  Personally I like these little details!  They provide extra context as to how the software was developped in a way that is more accessible for non-programmers.

SuaYoo replied:
Gotcha, yeah I can see the value in that. If this is aimed towards non-developers we should summarize the mechanism directly instead of linking to a github page. Especially for pixelmatch, the README is very technical.

How about something like:
```suggestion
Screenshots are compared by measuring the perceived difference between color samples and by the intensity of difference between pixels. These metrics are provided by the open-source tool [Pixelmatch](https://observablehq.com/@mourner/pixelmatch-demo).
```

SuaYoo commented on docs/user-guide/review.md:
```diff
@@ -0,0 +1,48 @@
+
+## Finish Review
+
+Once a satisfactory amount of pages have been reviewed, press the _Finish Review_ button to give the archived item an overall quality score ranging from "Excellent!" to "Bad". Update the description with any additional notes or considerations.
```
We should note that updating the description will update the archived item description:

```suggestion
Once a satisfactory amount of pages have been reviewed, press the _Finish Review_ button to give the archived item an overall quality score ranging from "Excellent!" to "Bad". You can add any additional notes or considerations in the archived item description, which can be edited during this step.
```

Shrinks99 committed:
Update docs/user-guide/review.md

Co-authored-by: sua yoo <sua@webrecorder.org>

Shrinks99 committed:
Update docs/user-guide/review.md

Co-authored-by: sua yoo <sua@webrecorder.org>

Shrinks99 committed:
Update docs/user-guide/archived-items.md

Co-authored-by: Ilya Kreymer <ikreymer@users.noreply.github.com>

Shrinks99 committed:
Improve clarity on viewer permissions

- Removes section count (not relevant)
- Adds a note in the permission levels section about crawl review

Shrinks99 committed:
Adds Sua's suggestion for keyword surfacing

Changes some of the language slightly from their comment.

Shrinks99 commented:
Thanks for the suggestions!  Have applied most of them and made a few edits for the two I didn't apply directly.  Should be ready to go?!

tw4l reviewed (approved):
I was waiting til the editing calmed down, but looks great, thanks!

SuaYoo reviewed (approved):
Woohoo!
